### PR TITLE
Added receive thread and mochad 10 simulator

### PR DESCRIPTION
**Pull request by Jakey69**
* The binding now also handles commands received from the Mochad X10 host. 
* Added a Mochad X10 simulator that can be used to test messages received from the host and to test broken pipe scenario's
* Added 'dim' functionality

### CODE REVIEW

**Comment by Jakey69** [comment]
Ready for review!

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10BrightCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		
+		eventPublisher.postCommand(itemName, new PercentType(level));
+	}
+
+	@Override
+	public String toString() {
+		return "MochadX10BrightCommand [value=" + value + ", toString()="
+				+ super.toString() + "]";
+	}
+}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10BrightCommand.java]: please add new line

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		
+		eventPublisher.postCommand(itemName, new PercentType(level));
+	}
+
+	@Override
+	public String toString() {
+		return "MochadX10DimCommand [value=" + value + ", toString()="
+				+ super.toString() + "]";
+	}
+}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java]: please add new line

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.openhab.core.library.types.PercentType;
+
+/**
+ * Implementation of the X10 Dim command
+ * 
+ * @author Jack Sleuters
+ * @since  1.7.0
+ *
+ */
+public class MochadX10DimCommand extends MochadX10Command {
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java]: the Dim- and the BrightCommand looks pretty similar. Any chance to extract code into a Superclass?

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/internal/MochadX10Binding.java [hunk 4]
```diff
@@ -81,9 +232,73 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
 					throw new ConfigurationException(port, "The specified port \"" + port + "\" is not a valid port number.");
 				}
 			}
+			
+			initializeBinding();
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/internal/MochadX10Binding.java]: please issues a disconnect when this Binding is deactivated() (you'll find a proper callback method for that)

**Comment by teichsta** [comment]
Thanks @Jakey69 for this contribution. Please find my review comments inline. Please also move the Simulator and its MainClass into the "simulator" directory which should be excluded from resulting jar (see build.properties for details). Please finally squash your commits into one after incorporating the Feedback. Best, Thomas E.-E.

**Comment by Jakey69** [comment]
Thanks for your Feb

**Comment by Jakey69** [comment]
Thanks for your feedback. I'll incorporate your suggestions. I'll let you know when I'm finished!

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10BrightCommand.java (hunk 1 above)
Comment (Jakey69) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10BrightCommand.java]: Done

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java (hunk 2 above)
Comment (Jakey69) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java]: Done

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java (hunk 3 above)
Comment (Jakey69) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/commands/MochadX10DimCommand.java]: I would like to have the difference between the Dim and Bright command in the different classes. Else, I would introduce some property which would state whether the comman is dim not bright, less OO in my opinion.

File: bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/internal/MochadX10Binding.java (hunk 4 above)
Comment (Jakey69) [inline bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/internal/MochadX10Binding.java]: Done

**Comment by Jakey69** [comment]
Moved this pull request to #2403.

**Pull request closed without merging**